tests/test_irdl.py

```diff
@@ -0,0 +1,243 @@
+"""Unit tests for IRDL."""
+
+from dataclasses import dataclass
+import pytest
+
+from xdsl.ir import Attribute, Data, ParametrizedAttribute
+from xdsl.irdl import AllOf, AnyAttr, AnyOf, AttrConstraint, BaseAttr, EqAttrConstraint, ParamAttrConstraint, ParameterDef, irdl_attr_definition
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.utils.exceptions import VerifyException
+
+
+@irdl_attr_definition
+class BoolData(Data[bool]):
```

Comment:
Would any of those classes be useful in the core xdsl?

Reply:
I would like as much as possible to not rely to xDSL dialects in our tests, so I would prefer keeping these here to make the tests self-contained, or possibly in a test dialect.
Since we replicate MLIR dialects, I would like to not add these `BoolData` to the dialects if possible, until we decide if we want new dialects.

If we want to make a test dialect, we should do it in a separate commit, but I would be worried about doing that because it would make our tests less self-contained, if that makes sense!

Reply:
Sure, thanks